### PR TITLE
DEV: Update dependencies and Docker image

### PR DESCRIPTION
**Pull request by trallard**
- DEV: Bump dependencies - ensure they match `doc-requirements.txt`
-  DEV: Also pins Python version to avoid issues during build in Gitpod
- DEV: Update mambaforge version

This is a PR aimed to support fixing the issue in #20346

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -21,13 +21,14 @@ dependencies:
...
-  - sphinx=4.1.1
+  - sphinx=4.2.0
+  - sphinx-panels
   - numpydoc=1.1.0
   - ipython
   - scipy
   - pandas
   - matplotlib
-  - pydata-sphinx-theme
+  - pydata-sphinx-theme=0.7.2
```
Comment (melissawm) [inline environment.yml:31]: Any specific reason for pinning this?

**Comment by melissawm** [comment]
We'll probably have to update the pydata-sphinx-theme soon to get the version switcher but otherwise looks good, so I'll go ahead and merge. Thanks @trallard !

**Comment by lucascolley** [comment]
> DEV: Also pins Python version to avoid issues during build in Gitpod

@trallard can you remember what issues were occurring? I wonder whether we can unpin this now, the version has been bumped up to Python 3.12 but it is still pinned.

**Comment by trallard** [comment]
I cannot remember (it was more than three years ago). 

> I wonder whether we can unpin this now, the version has been bumped up to Python 3.12 but it is still pinned.

I generally favour some pinning or upper/lower bounding for reproducibility, so that would be my same recommendation anyway. I am unsure if there are any reasons why you'd want to remove this, also I thought gitpod was no longer used in numpy.

**Comment by lucascolley** [comment]
okay, thanks!

**Pull request merged**